### DISCUSSION TITLE
Possibility to enable / disable filters

### DISCUSSION THREAD

**Discussion by edwinvdpol** [rappasoft/laravel-livewire-tables#2032] [Feature Requests]
Somewhat related to [this](https://github.com/rappasoft/laravel-livewire-tables/discussions/759) discussion.

I would like to see that you can enable / disable filters.
If the filter value of a disabled filter is set, it should act as **if** it is set (aka should be used in the builder etc.), and this value should be selected / filled by default.

It should apply the `disabled` attribute to the HTML input / select.

I have published all the filter views to add this to the existing filters.

I hope I am not the only one who would like this feature. 😉 

Thanks!

**Comment by lrljoe** [comment]
There are the following methods to hide a filter:
[#hiddenFromMenus](https://rappasoft.com/docs/laravel-livewire-tables/v3/filters/available-methods#content-hiddenfrommenus)
[#hiddenFromPills](https://rappasoft.com/docs/laravel-livewire-tables/v3/filters/available-methods#content-hiddenfrompills)
[#hiddenFromFilterCount](https://rappasoft.com/docs/laravel-livewire-tables/v3/filters/available-methods#content-hiddenfromfiltercount)
[#hiddenFromAll](https://rappasoft.com/docs/laravel-livewire-tables/v3/filters/available-methods#content-hiddenfromall)

What additionally were you thinking?

**Reply by edwinvdpol** [reply]
These methods will *hide* the filter without disabling the filter itself. I have a table property where you can define a kind of "preset".

For example, consider an invoice table with three boolean filters: `paid`, `open`, and `to_late` (where `open` and `to_late` represent payable statuses, and `paid` is non-payable). 

I’ve created a "payable" preset that sets the `paid` filter to `No`, and `to_late` and `open` filters to `Yes`. When this preset is active, users cannot modify these filters on the frontend. However, I still want them visible in the filter list so users can see that these filters are applied.

To achieve this, I add a `disabled` property while looping through the filters when rendering the component. If this property is set, the filter is displayed but cannot be modified on the frontend.

```html
<select name="..." disabled />
```

Does that make sense?

**Reply by lrljoe** [reply]
That does make sense.

It'll probably come in properly when I add in the feature to set custom attributes on a filter input, which is in the works, I just need to ensure that nothing is breakable (i.e. only permit certain attributes to merge)

For now - I do something similar with a property, apply the behaviour, and add a custom view into the toolbar configurableArea that gives a relevant message

**Comment by lrljoe** [comment]
For clarity, this is really more a "locked" filter, rather than a disabled one.

Worth noting, while you may disable the form input, a savvy user could still use an event via their browser.

What I'm thinking, is the ability to lock a filter, so that it prevents the value from changing from the one that was initially set, full stop.  But also presents a clean UX

It'll come after the next larger release to enable customising filter input attributes (which should help you out with your current disabled approach)

**Comment by lrljoe** [comment]
So to give an update.  There's an imminent feature for setting attributes on the filter input field, this should provide this feature at least to an extent.
